libvips/meson.build

```diff
@@ -0,0 +1,70 @@
+libvips_includedir = include_directories('include')
+libvips_deps = [
+    glib_dep,
+    gio_dep,
+    expat_dep,
+    gsf_dep,
+    fftw_dep,
+    magick_dep,
+    config_dep
+]
+
+libvips_sources = []
+subdir('include/vips')
+subdir('foreign')
+subdir('deprecated')
+subdir('arithmetic')
+subdir('resample')
+subdir('colour')
+subdir('conversion')
+subdir('convolution')
+subdir('freqfilt')
+subdir('histogram')
+subdir('draw')
+subdir('iofuncs')
+subdir('morphology')
+subdir('module')
+subdir('mosaicing')
+subdir('create')
+
+libvips_lib = library('vips',
+    enumtypes,
+    link_with: [
+        deprecated_lib,
+        foreign_lib,
```

Comment:
I think you probably want link_whole here. The static libraries need to be merged into a shared library, which is the purpose of `-Wl,--whole-archive`, otherwise with the default library being `shared` libvips.so will discard all symbols that only appear in archives to `link_with` but aren't referenced in the library's direct object files.

Also you could append each library to an array and avoid some use of placeholder arrays for optional components. :) (Also, it is more easily extended from a single location.)

Reply:
Thanks for the explanation, i've now adapted it to your suggestion